### ISSUE TITLE
Pylint prints error about cache files in fresh environment with pylint 2.10.2

### ISSUE THREAD

**Issue by GeZZu** [PyCQA/pylint#4900]
### Bug description

When I run pylint for my code in fresh environment (in build server), pylint tells me that it's not able to create cache directory (and files) to ``%localappdata%\pylint\pylint\Cache (Unable to create directory [full path]``).

Nevertheless I get the expected linting results written to a file.

I was able to reproduce the issue in my environment by upgrading pylint to 2.10.2 and removing pylint folder from ``%localappdata% and .pylint.d folder from %userprofile%``

If I have .pylint.d folder in ``%userprofile%``, I'll get message, which tells that .pylint.d is obsolete and ``PYLINTHOME`` now points to ``%localappdata%\pylint\pylint\Cache``. In that case the cache files are stored properly to ``PYLINTHOME``.

Everything also works in the fresh environment if I create the cache folder ``%localappdata%`` before running pylint.

### Command used

```shell
pylint file.py
```

### Pylint output

```shell
Unable to create directory
```

### Expected behavior

No error messages

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.9.6 (tags/v3.9.6:db3ff76, Jun 28 2021, 15:26:21) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by forana** [comment]
Seeing this as well - this has broken our CI.

**Comment by Pierre-Sassoulas** [comment]
I wonder if we should try to create a cache in the local directory if we can't access the ``XDG_CACHE``, in continuous integration we can't expect write access to the standard cache, so maybe using the local dir would be better.

**Comment by GeZZu** [comment]
I don't know how things have been implemented under the hood, but I would assume that write access is not the issue, you just need to verify that the cache folder exists and create it and all it's parents if it doesn't exist before trying to write anything to there.

I believe that the "Unable to create directory" message comes as it tries to create only the Cache folder, but not it's parents, which are also missing.

It also seems that the cache files are properly written to Cache folder if .pylint.d folder exists. So somewhere the code is cabable of being able to create the Cache folder and all it's parent's properly.

Anyway I'm totally fine with local cache dir also, if the normal cache dir permissions doesn't allow accessing it.

**Comment by DanielNoord** [comment]
Based on a @GeZZu's comment and my own testing I assumed changing the function to `os.makedirs` instead of `os.mkdir` could solve the issue as the former also creates any necessary parent directories.
From my testing this seems to solve the issue, but I can't test on a windows computer so somebody else might also want to check this!

**Comment by GeZZu** [comment]
Thanks.
Your change fixed it.

**Comment by DanielNoord** [comment]
Good to hear! Going to make a PR!

**Comment by trampgeek** [comment]
OK, so that's not the problem.

I've never seen this error myself, though I see there was a bug with pylint back in 2021 that caused such error messages.

What's the output from the following two commands, please?

1. pylint --version
2. lsb_release -a

And to confirm, before we waste more time on this: do you actually need pylint?
